src/include/duckdb/main/appender.hpp

```diff
@@ -114,12 +123,18 @@ class BaseAppender {
 };
 
 class Appender : public BaseAppender {
-	//! A reference to a database connection that created this appender
+	//! A shared pointer to the context of this appender.
 	shared_ptr<ClientContext> context;
-	//! The table description (including column names)
+	//! The table description including the column names.
 	unique_ptr<TableDescription> description;
-	//! The default expressions
-	unordered_map<idx_t, Value> default_values;
+	//! All table default values.
+	unordered_map<column_t, Value> default_values;
+
+	//! If not empty, then this holds all columns provided by the appender.
+	//! Any other columns default to NULL, or their default values.
+	vector<string> active_columns;
+	//! The active default values.
+	unordered_map<column_t, Value> active_default_values;
```

Comment:
Instead of having all of these extra fields, maybe we can have a `vector<LogicalIndex> column_ids` similar to how we handle this in other places in the code? We can then look-up the column names, types default values using those column-ids when we need them.

Reply:
In the case of `active_default_values ` and `Appender::AppendDefault();`, that would mean we need to perform a lookup for each appended value - maybe that is impacting performance too much? For the `active_columns`, I have no preference - I can turn it into a `vector<LogicalIndex> column_ids`. :)

Reply:
I don't quite follow - both `default_values` and `active_default_values` are `unordered_maps` - so we would need to do the look-up in a map in any case, no?

I think the more efficient design is to make `default_values` a `vector<Value>`, then use the `column_ids` to index into that vector.

Reply:
Otherwise I think removing `active_columns` and `active_types` and using the `column_ids` is a good idea

Reply:
> Otherwise I think removing active_columns and active_types and using the column_ids is a good idea

Ah yes I had removed my comment again because I agreed after a second look - sorry!